hermes/probe/create/create.go

```diff
@@ -35,85 +38,84 @@ import (
+	FileNameFormat          = "Hermes_%02d_%x"
+	minFileID               = 1
+	maxFileID               = 50
+	maxFileSizeBytes        = 1000
 	hermesAPILatencySeconds = "hermes_api_latency_s"
 )
 
 type randomFile struct {
-	id   int32 // id is a positive integer in the range [begin,end]
-	size int   // size is the file size in bytes.
+	id        int32
+	sizeBytes int
 }
 
 type randomFileReader struct {
-	size int // size in bytes
-	i    int // currently reading this byte
+	sizeBytes int
+	// currently reading this byte
+	i    int
 	rand *rand.Rand
 }
 
-// Read implements the standard Read interface, it reads data from a randomFileReader.
+// Read implements the standard Read interface, it returns a random stream of bytes generated by a pseudo-random generator.
 // Argument:
-//      buf: a byte slice serves as a buffer
+//      buf: a byte slice serves as an output buffer
 // Returns:
-//	n:  the size of the buffer buf following a Read operation
-//	err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
+//		n:  the number of bytes read
+//		err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
 func (r *randomFileReader) Read(buf []byte) (n int, err error) {
 	// check whether Read is done
-	if r.i >= r.size {
+	if r.i >= r.sizeBytes {
 		return 0, io.EOF
 	}
 	b := buf
-	if len(buf) > (r.size - r.i) {
-		// if the length of buffer is greater than the number of bytes left to read make the size of the buffer match the number of bytes left to read
-		b = buf[:(r.size - r.i)]
+	if len(buf) > (r.sizeBytes - r.i) {
+		// if the length of buffer is greater than the number of bytes left to read make the sizeBytes of the buffer match the number of bytes left to read
+		b = buf[:(r.sizeBytes - r.i)]
 	}
-	n, err = r.rand.Read(b) // n is now the length  of the buffer
+	// n is now the length  of the buffer
+	n, err = r.rand.Read(b)
 	if err != nil {
-		return n, err // in this case n = 0
+		// in this case n = 0
+		return n, err
 	}
 	r.i += n
 	return n, err
 }
 
-// NewReader creates a new randomFileReader and returns a pointer to it
-// Returns:
-//	*randomFileReader: a Reader that supports randomFiles. This Reader contains information about the file size in bytes as well as a pseudo-random bytes generator
-// Warning: The randomFileReader produced by NewReader is not thread safe.
-func (f *randomFile) NewReader() *randomFileReader {
+func (f *randomFile) newReader() *randomFileReader {
 	//  id will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
 	return &randomFileReader{
-		size: f.size,
-		rand: rand.New(rand.NewSource(int64(f.id))), // rand.NewSource() requires an int64 as an argument
+		sizeBytes: f.sizeBytes,
+		rand:      rand.New(rand.NewSource(int64(f.id))),
 	}
 }
 
 func newRandomFile(fileID int32, fileSize int) (*randomFile, error) {
```

Comment:
Nit: The `file` prefix in the arguments are not necessary. `id` and `size` (or preferably `sizeBytes`) would do just as well, as the reader already knows that it's about the file.

Reply:
changed